### PR TITLE
pirate content

### PR DESCRIPTION
**Pull request by brainfood1183**
Contains new pirate clothing and equipment to improve the pirate antag experience. A new fun chest that cargo can (waste credits on) order which contains fun pirate cosplay. Alterations to the PirateGear spawner which will give pirates new pirate clothing and equipment listed below (this will also alter the starting gear of pirates during a pirate antag event).

untitled-168

untitled-177

Includes

- Pirate cannon - Hits like c4 against structures, does little to no tile damage and delivers about 80 damage to crewmembers. One shot and reload. I expect pirates to have 1 or 2 of these and maybe 3 or 4 extra cannonballs.
-  Cannonball - ammunition for the cannon, should be 3 or 4 per ship.
-   Cutlass - a melee weapon for pirates, does 18 damage with a regular swing speed (captains sabre is a lot better than cutlass).
-   Energy Cutlass - comparable to an energy sword except slightly different damage outputs.
-   Foam Cutlass - a foam version of a cutlass for dressing up as a pirate and having fun.
-   Pirate Slops - a pirate jumpsuit.
-   Pirate accent - triggered by wearing Pirate Slops makes you sound like a pirate by swapping out words with more piratey words.
- Pirate Tricord - A new pirate hat, typically worn by the first mate.
-   Deep Space EVA Suit (softsuit) - replaces the standard EVA suit for pirates and could be worn by crewmembers (looks like deep diver costume and should replace the spawns on the pirate ship). It is heavy and you move slowly wearing it so idea is pirates will take off the EVA suit when they board station. Comes with a hardsuit helmet.
-   Pirate Captains Cloak - a cloak to help the pirate captain stand out from the other pirates.
-   Pirate Captain Ancient Hardsuit - a hardsuit for the pirate captain with built in helmet, similar to the captains hardsuit but slightly worse.
- Blunderbuss - shotgun which fires one shot.
-   Pirate chest - comes in 4 varieties, empty, antag - captain which is designed to be placed in captains quarters of pirate ship, antag -crew which is designed to go in the armory of the pirateship and fun which is a fun pirate themed crate that cargo can order. The pirate themed crate contains 2 outfits to dressup 2 crew members and 2 foam cutlass swords.
    Alterations to PirateGear- the PirateGear loadout

1.  swapped brown jumpsuit for pirate slops
2.  swapped pirate captain hat for black bandana (pirate captain hats still on pirate ship)
3.  removed pirate garb (pirate garbs still on the pirate ship)
4.   swapped the engineering backpack for a brown leather satchel which is filled and contains a survival box, cutlass, pirate revolver and eyepatch.
5.  swapped workboots for lacedshoes
6.  pirategear still keeps the appraisal tool, passenger PDA and utility belt.

this pirate gear is used by the pirate spawner so pirate that spawn will now have this loadout.
i also created a piratecaptaingear set and piratefirstmate set, however there is currently no spawner for a piratecaptain or first mate.

Pirate ship has been updated as well.

- Redesigned the look of the ship slightly.
- Added scrubbers to remove bad air from ship.
- Changed the loadout of the pirate ship and placed in the new pirate equipment.

What should be done after this pr

- The blunderbuss currently is a 1 shot firing shotgun, could possibly use a rebalance at some point.

:cl: brainfood

- Added new pirate clothing and equipment including cannons, hardsuits, cutlass and more.
- Changed the pirate shuttle to take advantage of the new equipment.

### CODE REVIEW

**Review by brainfood1183 — COMMENTED** [review]
small update

File: Content.Server/Speech/EntitySystems/PirateAccentSystem.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+public sealed class PirateAccentSystem : EntitySystem
+{
+    private static readonly Dictionary<string, string> DirectReplacements = new()
+    {
+        { "hi", "Arghh" },
+        { "hello", "Ahoy" },
+        { "hey", "Gar" },
+        { "yo", "Yarr" },
+        { "friend", "matey" },
+        { "crew", "me harties" },
```
Comment (GoodWheatley) [inline Content.Server/Speech/EntitySystems/PirateAccentSystem.cs:16]: believe this should be "me hearties"

File: Resources/Prototypes/Catalog/Fills/Crates/antag.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  suffix: antag
+  parent: CratePirate
+  components:
+  - type: StorageFill
+    contents:
+      - id: HatBandRed
+        amount: 1
+      - id: Cutlass
+        amount: 1
+      - id: IngotGold
```
Comment (GoodWheatley) [inline Resources/Prototypes/Catalog/Fills/Crates/antag.yml:12]: starting with gold seems kind of strange, considering that its just free money for the pirates

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/e_sword.yml [hunk 3]
```diff
@@ -106,3 +106,31 @@
...
+  suffix: E-Cutlass
+  description: An exotic energy weapon.
+  components:
+  - type: EnergySword
+    secret: true
+    litDamageBonus:
+        types:
+            Slash: 10.5
+            Heat: 10.5
+            Blunt: -1
```
Comment (GoodWheatley) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/e_sword.yml:123]: This is just a straight upgrade over the regular esword. Maybe lower the attack speed to balance it?

File: Content.Server/Speech/EntitySystems/PirateAccentSystem.cs (hunk 1 above)
Comment (brainfood1183) [reply Content.Server/Speech/EntitySystems/PirateAccentSystem.cs:16]: yea will change it

File: Resources/Prototypes/Catalog/Fills/Crates/antag.yml (hunk 2 above)
Comment (brainfood1183) [reply Resources/Prototypes/Catalog/Fills/Crates/antag.yml:12]: good point, not exactly sure what could be in a pirate chest

File: Content.Server/Speech/EntitySystems/PirateAccentSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/Speech/EntitySystems/PirateAccentSystem.cs:16]: None of these are localised either.

Also isn't there a replacementaccentcomponent for straight up replacements?

File: Content.Server/Speech/EntitySystems/PirateAccentSystem.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+        SubscribeLocalEvent<PirateAccentComponent, AccentGetEvent>(OnAccentGet);
+    }
+    private void OnAccentGet(EntityUid uid, PirateAccentComponent component, AccentGetEvent args)
+    {
+        foreach (var (first, replace) in DirectReplacements)
+        {
+            args.Message = Regex.Replace(args.Message, $@"(?<!\w){first}(?!\w)", replace, RegexOptions.IgnoreCase);
+        }
+    }
```
Comment (metalgearsloth) [inline Content.Server/Speech/EntitySystems/PirateAccentSystem.cs:107]: Seems very inefficient to iterate 100 words every message?

File: Resources/Prototypes/Entities/Clothing/Head/helmets.yml [hunk 5]
```diff
@@ -294,3 +294,16 @@
         Slash: 0.95
         Heat: 0.90
         Radiation: 0.75
+
+- type: entity
+  parent: ClothingHeadEVAHelmetBase
+  id: ClothingHeadHelmetCosmonaut
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Clothing/Head/helmets.yml:300]: Why's it called cosmonaut.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/explosives.yml [hunk 6]
```diff
@@ -124,3 +124,17 @@
...
+  suffix: pirate
+  parent: BaseGrenade
+  components:
+  - type: CartridgeAmmo
+    proto: BulletCannonBall
+    deleteOnSpawn: true
+  - type: Sprite
+    netsync: false
+    sprite: Objects/Weapons/Guns/Projectiles/projectiles2.rsi
+    state: ball
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/explosives.yml:140]: Does this really need a separate cartridge

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml [hunk 7]
```diff
@@ -144,3 +144,43 @@
...
+      ballistic-ammo: !type:Container
+        ents: []
+    type: ContainerContainer
+  - type: Clothing
+    sprite: Objects/Weapons/Guns/Launchers/pirate_cannon.rsi
+  - type: Gun
+    fireRate: 1
+    selectedMode: FullAuto
+    availableModes:
+      - FullAuto
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml:171]: SemiAuto?

File: Resources/Textures/Objects/Weapons/Guns/Launchers/pirate_cannon.rsi/meta.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		"name": "inhand-right",
+		"directions": 4
+	},
+    {
+      "name": "equipped-BACKPACK",
+      "directions": 4
+    }
+    ]
+}
```
Comment (metalgearsloth) [inline Resources/Textures/Objects/Weapons/Guns/Launchers/pirate_cannon.rsi/meta.json:26]: Please don't mix spaces and tabs like this, configure your IDE to just convert it all to spaces.

**Review by metalgearsloth — DISMISSED** [review]
1. The capes look weird, like they're highlighted and I don't imagine it plays well with interactionoutline
2. The cutlasses and backpack sprites are enormous, teh cutlass is bigger than mobs.
3. The cannon back sprite doesn't look very good, like it's completely straight and just looks like a gradient was tossed on it.
4. The straight horizontal lines on the clothing is very eh too.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml (hunk 7 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml:171]: only has 1 shot so it doesnt make a huge difference if its auto or semiauto

File: Resources/Prototypes/Entities/Clothing/Head/helmets.yml (hunk 5 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Clothing/Head/helmets.yml:300]: no idea, i didnt make this item

**Comment by AJCM-git** [comment]
will you come back to this? you can close it if you are busy

**Comment by brainfood1183** [comment]
> will you come back to this? you can close it if you are busy

I do plan to return to this although dont feel much drive at the moment. Pirates are almost never seen. That being said i do plan to work on it and have it done by the ned of feb.

**Comment by Cheackraze** [comment]
I have pirate shuttle fixes in the works, if this content were added along with the pirate shuttle fixes, it would be easier to design a game mode for them.

**Comment by metalgearsloth** [comment]
Re-open when ready.

File: Resources/Prototypes/Entities/Clothing/Head/helmets.yml (hunk 5 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Clothing/Head/helmets.yml]: this was fixed

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml (hunk 7 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml]: this was fixed

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/e_sword.yml (hunk 3 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/e_sword.yml]: cutlass does 16 dam and the e-cutlass same as the e-sword

**Comment by brainfood1183** [comment]
> 1. The capes look weird, like they're highlighted and I don't imagine it plays well with interactionoutline
> 
>     2. The cutlasses and backpack sprites are enormous, teh cutlass is bigger than mobs.
> 
>     3. The cannon back sprite doesn't look very good, like it's completely straight and just looks like a gradient was tossed on it.
> 
>     4. The straight horizontal lines on the clothing is very eh too.

these issues you had should all hopefully be sorted now. New cape, cutlass, jumpsuit and cannon sprites and the backpack was removed.

File: Content.Server/Speech/EntitySystems/PirateAccentSystem.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Content.Server.Speech.Components;
+
+namespace Content.Server.Speech.EntitySystems;
+
+public sealed class PirateAccentSystem : EntitySystem
+{
+    private static readonly Dictionary<string, string> DirectReplacements = new()
```
Comment (DrSmugleaf) [inline Content.Server/Speech/EntitySystems/PirateAccentSystem.cs:9]: Petition to add station => docks

**Review by DrSmugleaf — APPROVED** [review]

File: Content.Server/Speech/EntitySystems/PirateAccentSystem.cs [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        { "pirate", "buccaneer" },
+        { "song", "sea shanty" },
+        { "stop", "avast" },
+        { "go", "heave ho" },
+        { "understood", "savvy" },
+        { "drunk", "squiffy" },
+        { "you", "ya" },
+        { "shuttle", "ship" },
+        { "yea", "aye" },
+        { "wait", "hold" },
```
Comment (DrSmugleaf) [inline Content.Server/Speech/EntitySystems/PirateAccentSystem.cs:29]: These aren't localized as Sloth said. Look at how accent prototypes do it.

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

**Comment by AJCM-git** [comment]
@brainfood1183 you coming back to this?

**Comment by brainfood1183** [comment]
> @brainfood1183 you coming back to this?

yea i plan to finish this next.

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml [hunk 11]
```diff
@@ -668,3 +668,60 @@
...
+  - type: ExplosionResistance
+    damageCoefficient: 0.6
+  - type: ToggleableClothing
+    clothingPrototype: ClothingHeadHelmetHardsuitPirateCap
+
+- type: entity
+  parent:  ClothingOuterHardsuitBase
+  id:  ClothingOuterHardsuitPirateEVA
+  name: deep space EVA suit
+  suffix: pirate
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:706]: .

File: Resources/Prototypes/Roles/Jobs/Fun/misc_startinggear.yml
Comment (AJCM-git) [inline Resources/Prototypes/Roles/Jobs/Fun/misc_startinggear.yml:1]: Move pirate stuff to a new file now that there are more

File: Resources/Textures/Clothing/Head/Hardsuits/piratecaptainhelm.rsi/meta.json [hunk 12]
```diff
@@ -0,0 +1,18 @@
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "created by /Discord: brainfood#7460/github: brainfood1183/ for space-station-14.",
```
Comment (AJCM-git) [inline Resources/Textures/Clothing/Head/Hardsuits/piratecaptainhelm.rsi/meta.json:4]: same for the 4 or so others that dont follow this
```suggestion
    "copyright": "Made by brainfood1183 (Github) for ss14",
```

File: Resources/Textures/Structures/Storage/Crates/piratechest.rsi/meta.json [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+            "flags": {}
+        },
+        {
+            "name": "welded",
+            "select": [],
+            "flags": {}
+        },
+        {
+            "name": "sparking",            "delays": [[0.1,0.1,0.1,0.1,0.1,0.1]]
+        },
```
Comment (AJCM-git) [inline Resources/Textures/Structures/Storage/Crates/piratechest.rsi/meta.json:32]: format delays correctly (should be done just fine by your IDE but if it doesnt work there are a lot of examples around of how it should look)

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Add a screenshot of the shuttle on your pr

**Review by AJCM-git — DISMISSED** [review]
i dont see a changelog

**Comment by brainfood1183** [comment]
Hopefully everything should be covered now.

**Comment by CaptainSqrBeard** [comment]
this is now most oldest PR that still not merged

**Comment by brainfood1183** [comment]
> this is now most oldest PR that still not merged

Yea, its famously old.

**Comment by Froozigiusz** [comment]
The current sprites are such a glow up from the first version, they also look very nice by themselves.

With the cannon balls not being able to pierce the ground, I think they would be even more fun to fight against, as many civilians aren't able to survive the spaced corridors. But I wonder how numerous would they be in order to be effective, if melee is their preferred choice of weapon type.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**